service/src/main/java/tds/session/repositories/ExternRepository.java

```diff
@@ -0,0 +1,14 @@
+package tds.session.repositories;
+
+import java.util.Optional;
+
+import tds.session.Extern;
+
+public interface ExternRepository {
+    /**
+     * Fetches the extern by given client name
+     * @param clientName client name for extern
+     * @return optional containing {@link tds.session.Extern Extern} otherwise empty Optional
+     */
+    Optional<Extern> getExternByClientName(String clientName);
```

Comment:
We should discuss this in more detail in today's parking lot.  500 may make sense here but then what should the caller do when it receives that response?  Throw a 500 as well?

And example on how the system will work with what is being proposed: Student asks the exam service to open an exam leading to exam calling this code to get the extern.  So for some reason the extern (view not the _externs table) isn't populated and exam gets a 500.  Should it throw a 500 or handle it returning a different error back to student?  For this example assume there is no default handling that can be done.